src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ICoalesceAssignmentOperation.cs

```diff
@@ -0,0 +1,895 @@
+  Target: 
+    IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid) (Syntax: 'o1')
+      Children(0)
+  WhenNull: 
+    IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid) (Syntax: 'o2')
+      Children(0)
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(6,19): error CS0103: The name 'o1' does not exist in the current context
+                //         /*<bind>*/o1 ??= o2/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "o1").WithArguments("o1").WithLocation(6, 19),
+                // file.cs(6,26): error CS0103: The name 'o2' does not exist in the current context
+                //         /*<bind>*/o1 ??= o2/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "o2").WithArguments("o2").WithLocation(6, 26)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<AssignmentExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void CoalesceAssignment_AsExpression()
+        {
+
+            string source = @"
+class C
+{
+    static void M(object o1, object o2)
+    {
+        /*<bind>*/M2(o1 ??= o2)/*</bind>*/;
+    }
+    static void M2(object o) {}
+}
+";
+            string expectedOperationTree = @"
+IInvocationOperation (void C.M2(System.Object o)) (OperationKind.Invocation, Type: System.Void) (Syntax: 'M2(o1 ??= o2)')
+  Instance Receiver: 
+    null
+  Arguments(1):
+      IArgumentOperation (ArgumentKind.Explicit, Matching Parameter: o) (OperationKind.Argument, Type: null) (Syntax: 'o1 ??= o2')
+        ICoalesceAssignmentOperation (OperationKind.CoalesceAssignment, Type: System.Object) (Syntax: 'o1 ??= o2')
+          Target: 
+            IParameterReferenceOperation: o1 (OperationKind.ParameterReference, Type: System.Object) (Syntax: 'o1')
+          WhenNull: 
+            IParameterReferenceOperation: o2 (OperationKind.ParameterReference, Type: System.Object) (Syntax: 'o2')
+        InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+        OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<InvocationExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void CoalesceAssignment_CheckedDynamic()
+        {
+
+            string source = @"
+class C
+{
+    static void M(dynamic d1, dynamic d2)
+    {
+        checked
+        {
+            /*<bind>*/d1 ??= d2/*</bind>*/;
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+ICoalesceAssignmentOperation(IsChecked: True) (OperationKind.CoalesceAssignment, Type: dynamic) (Syntax: 'd1 ??= d2')
```

Comment:
>IsChecked: True [](start = 29, length = 15)

It would be good to have explicit tests for unchecked dynamic case and both flavors for non-dynamic.

Reply:
No longer relevant, as `IsChecked` has been removed.

---
In reply to: [209402877](https://github.com/dotnet/roslyn/pull/29063#discussion_r209402877) [](ancestors = 209402877)